src/mono/mono/metadata/assembly.c

```diff
@@ -2501,6 +2501,16 @@ mono_assembly_open_from_bundle (MonoAssemblyLoadContext *alc, const char *filena
 	for (i = 0; !image && bundles [i]; ++i) {
 		if (strcmp (bundles [i]->name, is_satellite ? filename : name) == 0) {
 			image = mono_image_open_from_data_internal (alc, (char*)bundles [i]->data, bundles [i]->size, FALSE, status, refonly, FALSE, name);
+#if TARGET_WASM
```

Comment:
not sure if we need to keep this compatible but only do this on netcore?

```suggestion
#if defined(TARGET_WASM) && defined(ENABLE_NETCORE)
```